titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/endpoint/v3/grpc/DefaultJobManagementServiceGrpc.java

```diff
@@ -436,16 +441,29 @@ public void killJob(JobId request, StreamObserver<Empty> responseObserver) {
     @Override
     public void killTask(TaskKillRequest request, StreamObserver<Empty> responseObserver) {
         execute(callMetadataResolver, responseObserver, callMetadata -> {
+            Pair<com.netflix.titus.api.jobmanager.model.job.Job<?>, com.netflix.titus.api.jobmanager.model.job.Task> jobTaskPair = jobOperations.findTaskById(request.getTaskId()).orElse(null);
+            if (jobTaskPair == null) {
+                responseObserver.onError(JobManagerException.taskNotFound(request.getTaskId()));
+                return;
+            }
+
             String reason = String.format("User initiated task kill: %s", CallMetadataUtils.toReasonString(callMetadata));
-            jobOperations.killTask(request.getTaskId(), request.getShrink(), reason).subscribe(
-                    nothing -> {
-                    },
-                    e -> safeOnError(logger, e, responseObserver),
-                    () -> {
-                        responseObserver.onNext(Empty.getDefaultInstance());
-                        responseObserver.onCompleted();
-                    }
-            );
+            ReactorExt.toObservable(authorizationService.authorize(callMetadata, jobTaskPair.getLeft()))
+                    .flatMap(authorizationResult -> {
+                        if (!authorizationResult.isAuthorized()) {
+                            return Observable.error(new IllegalArgumentException(authorizationResult.getReason()));
```

Comment:
Do we need a different exception here? How do you expect to map these types of error to `Forbidden` and `PermissionDenied`?

Reply:
Good point. I will fix that.